### PR TITLE
qemu: Add -enable-kvm to force (attempted) use (or not) of KVM

### PR DESCRIPTION
**Pull request by ijc**
The default is set by probing for /dev/kvm. Use of KVM can be forcibly disabled
with `-enable-kvm=false`. Note that passing `-enable-kvm=true` when `/dev/kvm` is
not present still boots but complains:

    Could not access KVM kernel module: No such file or directory
    failed to initialize KVM: No such file or directory
    Back to tcg accelerator.

Signed-off-by: Ian Campbell <ian.campbell@docker.com>

**- What I did**

Added `-enable-kvm` option to `linuxkit run qemu`.

**- How I did it**

Plumbed in some new golang `flags` usage.

**- How to verify it**

Try passing `-enable-kvm` or `-enable-kvm=false` before and after renaming `/dev/kvm` to something else.

**- Description for the changelog**

LinuxKit run using QEMU backend can now opt-in/out to use of KVM. Default remains to autodetect.

**- A picture of a cute animal (not mandatory but encouraged)**
[](https://www.metal-archives.com/albums/A_Forest_of_Stars/Beware_the_Sword_You_Cannot_See/478451)

### CODE REVIEW

File: src/cmd/linuxkit/run_qemu.go [hunk 1]
```diff
@@ -68,6 +73,7 @@ func runQemu(args []string) {
 	fw := flags.String("fw", "/usr/share/ovmf/bios.bin", "Path to OVMF firmware for UEFI boot")
 
 	// VM configuration
+	enableKVM := flags.Bool("enable-kvm", haveKVM(), "Enable or disable QEMU")
```
Comment (justincormack) [inline src/cmd/linuxkit/run_qemu.go]: comment says "qemu" not kvm

Comment (ijc) [inline src/cmd/linuxkit/run_qemu.go]: Sigh. I keep making the same mistake. Will fix.

Comment (justincormack) [inline src/cmd/linuxkit/run_qemu.go]: may as well just call the flag `--kvm`

Comment (ijc) [inline src/cmd/linuxkit/run_qemu.go]: The "QEMU" option is `-enable-kvm` although I confess that wasn't a consideration when I typed it above and it doesn't have the same `=false` ability, so compat is probably not the biggest concern.

I'll change.

File: src/cmd/linuxkit/run_qemu.go [hunk 2]
```diff
@@ -68,6 +73,7 @@ func runQemu(args []string) {
 	fw := flags.String("fw", "/usr/share/ovmf/bios.bin", "Path to OVMF firmware for UEFI boot")
 
 	// VM configuration
+	enableKVM := flags.Bool("kvm", haveKVM(), "Enable or disable KVM acceleration")
```
Comment (justincormack) [inline src/cmd/linuxkit/run_qemu.go]: I think the comment should read like a boolean, ie "Enable KVM acceleration"

Comment (ijc) [inline src/cmd/linuxkit/run_qemu.go]: Done

**Pull request merged**